### PR TITLE
Add e-mail address validation method

### PR DESCRIPTION
**Pull request by AlexRukhlin**
EmailAddressParser - the new class that parses e-mal addresses according to [ RFC 5322 ](https://tools.ietf.org/html/rfc5322) (sections 3.2.3 and 3.4.1) with a more readable form given in the informational [RFC 3696](https://tools.ietf.org/html/rfc3696), plus the [associated errata](http://www.rfc-editor.org/errata_search.php?rfc=3696).

FeedbackDialog - uses that parser via TEE validators mechanism.

### CODE REVIEW

**Comment by AlexRukhlin** [comment]
The PR has been approved by @DavidStaheli

File: source/com.microsoft.tfs.util/tests/com/microsoft/tfs/util/email/EmailAddressParserTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+import com.microsoft.tfs.util.StringUtil;
+
+import junit.framework.TestCase;
+
+public class EmailAddressParserTest extends TestCase {
+    final EmailAddressParser parser = new EmailAddressParser();
+
+    public void testEmailAddressGoodSimple() {
+        final boolean result = parser.parse("JohnDoe@live.com"); //$NON-NLS-1$
```
Comment (olivierdagenais) [inline source/com.microsoft.tfs.util/tests/com/microsoft/tfs/util/email/EmailAddressParserTest.java]: Consider using [Reserved Top Level DNS Names](http://tools.ietf.org/html/rfc2606) in test code.

File: source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+EmailAddressParser.SubdomainCannotBeEmpty=The subdomain name cannot be empty.
+EmailAddressParser.SubdomainCannotEndWithHyphen=The subdomain name cannot end with the hyphen (''-'') character.
+EmailAddressParser.SubdomainCannotStartWithHyphen=The subdomain name cannot start with the hyphen (''-'') character.
+EmailAddressParser.SubdomainTooLong=The subdomain name cannot be longer than 63 characters.
+EmailAddressParser.TooManyDomainLevels=The domain part cannot more that 127 levels.
+EmailAddressParser.UnallowedCharacterFormat=Unallowed character ''{0}'' at position {1}.
+EmailAddressParser.UnexpectedEol=Unexpected end of string in the e-mail address.
+EmailAddressParser.UnexpectedEolInQuoteFormat=Unexpected end of string in the e-mail address. Probably the last quote mark at position {0} was not closed.
+EmailAddressParser.UnquotedAtInQuoteFormat=Unexpected unquoted ''@'' character inside of quoted substring at position {0}. Probably the last quote mark at position {1} was not closed.
+EmailAddressParser.WrongCaracterInSubdomain=The subdomain name can cosist only of alphanumeric characters ang hyphens (''-'').
```
Comment (olivierdagenais) [inline source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties]: Typographical error in `Caracter` (constant/property name), as well as message: `cosist` and `ang`.

File: source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/feedback/FeedbackDialog.java [hunk 3]
```diff
@@ -154,4 +199,37 @@ public String getComment() {
...
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected IValidity computeValidity(String text) {
+            if (emailErrorMessageLabel == null) {
+                /*
+                 * Nao all controls are created yet.
```
Comment (olivierdagenais) [inline source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/feedback/FeedbackDialog.java]: `Nao`?

File: source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties [hunk 4]
```diff
@@ -1,4 +1,24 @@
 AdvisoryFileLock.InterruptedAcquiringLockFormat=Interrupted while acquiring blocking lock on FileChannel for file {0}
+EmailAddressParser.AddressTooLong=The e-mail address cannot be longer that 256 characters.
+EmailAddressParser.DomainCannotEndWithDot=The domain part cannot end with the dot (''.'') character
+EmailAddressParser.DomainCannotStartWithDot=The domain part cannot start with the dot (''.'') character
+EmailAddressParser.DomainTooLong=The domain part cannot be longer than 253 characters.
+EmailAddressParser.DotAtBeginning=The dot (''.'') character is not allowed at the beginning of the local part
+EmailAddressParser.DotAtEnd=The dot (''.'') character is not allowed at the end of the local part
```
Comment (olivierdagenais) [inline source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties]: Not all messages end with a `.`.

File: source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties [hunk 5]
```diff
@@ -1,4 +1,24 @@
...
+EmailAddressParser.EmptyLocalpart=The local part cannot be empty.
+EmailAddressParser.LocalPartTooLong=The local part cannot be longer that 64 characters.
+EmailAddressParser.ManyDots=Multiple dot (''.'') characters are not allowed
+EmailAddressParser.MissingDomainpart=Missing domain part.
+EmailAddressParser.SubdomainCannotBeEmpty=The subdomain name cannot be empty.
+EmailAddressParser.SubdomainCannotEndWithHyphen=The subdomain name cannot end with the hyphen (''-'') character.
+EmailAddressParser.SubdomainCannotStartWithHyphen=The subdomain name cannot start with the hyphen (''-'') character.
+EmailAddressParser.SubdomainTooLong=The subdomain name cannot be longer than 63 characters.
+EmailAddressParser.TooManyDomainLevels=The domain part cannot more that 127 levels.
+EmailAddressParser.UnallowedCharacterFormat=Unallowed character ''{0}'' at position {1}.
```
Comment (olivierdagenais) [inline source/com.microsoft.tfs.util/src/com/microsoft/tfs/util/messages.properties]: Inconsistent wording in error messages.  Sometimes the subject is first (i.e. "The subdomain name..") and others, like this message, start with the quality of what is wrong (i.e. "Unallowed", "Unexpected").

File: source/com.microsoft.tfs.util/tests/com/microsoft/tfs/util/email/EmailAddressParserTest.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the repository root.
+
+package com.microsoft.tfs.util.email;
+
+import com.microsoft.tfs.util.StringUtil;
+
+import junit.framework.TestCase;
+
+public class EmailAddressParserTest extends TestCase {
```
Comment (olivierdagenais) [inline source/com.microsoft.tfs.util/tests/com/microsoft/tfs/util/email/EmailAddressParserTest.java]: I count 20 possible error messages and 8 tests of where the expected result is `false`.

**Pull request merged**